resources/enclosure.rb

```diff
@@ -32,3 +34,20 @@
 action :remove do
   remove
 end
+
+action :reconfigure do
+  item = load_resource
+  item.retrieve!
+  item.configuration
+end
+
+action :refresh do
```

Comment:
A good way to trigger a refresh would be using an attribute when calling a CHEF action that might need a refresh after the execution.
That can be done by the notify.

Regarding the reconfigure I agree with Thiago!